src/entity2/revision.ts

```diff
@@ -0,0 +1,55 @@
+import { Entity, PrimaryGeneratedColumn, Column, BaseEntity, JoinColumn, ManyToOne } from 'typeorm';
+
+// eslint-disable-next-line import/no-cycle
+import { Dataset } from './dataset';
+import { User } from './user';
+
+interface Revision {
+    id: string;
+    revision_index: number;
+    dataset: Dataset;
+    creation_date: Date;
+    previous_revision: Revision;
+    online_cube_filename: string;
+    publish_date: Date;
+    approval_date: Date;
+    approved_by: User;
+    created_by: User;
+}
+
+@Entity()
+export class RevisionEntity extends BaseEntity implements Revision {
+    @PrimaryGeneratedColumn('uuid')
+    id: string;
+
+    @Column({ type: 'int' })
+    revision_index: number;
+
+    @ManyToOne(() => Dataset, (dataset) => dataset.revisions)
+    @JoinColumn({ name: 'dataset_id' })
+    dataset: Dataset;
+
+    @Column({ type: 'timestamp', default: () => 'CURRENT_TIMESTAMP' })
+    creation_date: Date;
+
+    @ManyToOne(() => RevisionEntity, { nullable: true })
+    @JoinColumn({ name: 'previous_revision_id' })
+    previous_revision: Revision;
+
+    @Column({ type: 'varchar', length: 255, nullable: true })
+    online_cube_filename: string;
```

Comment:
I am not yet sure how many revisions of a particular dataset we can have online at a particular time (just the latest is the easiest and perhaps one extra for the current / most important draft), but yeah: it makes sense to store this in the revision rather than the dataset.

Reply:
When you create a new revision I rather assumed you'd create a new online cube... the old cubes would in theory still be available or at the very least archived off.